### ISSUE TITLE
Explain in error message that Type arguments are not *needed* with Protocol class when using type parameter syntax

### ISSUE THREAD

**Issue by wyattscarpenter** [microsoft/pyright#10449]
**Is your feature request related to a problem? Please describe.**

Today I was, mostly out of curiosity, adapting _typeshed.SupportsAdd to use type parameter syntax, and I ended up with this:

```py
from typing import Protocol

class Addable[Addend, Result](Protocol[Addend, Result]):
  def __add__(self, x: Addend, /) -> Result: ...
```

which gives the error:

```
Type arguments are not allowed with Protocol class when using type parameter syntax  (reportGeneralTypeIssues)
```

I googled around and eventually found https://peps.python.org/pep-0695/#type-parameter-declarations, upon the reading of which I discovered that the type arguments to Protocol here are actually *redundant*, and can be completely removed with no trouble. 

**Describe the solution you’d like**

The error message should mention this. Something like "Type arguments are not allowed with Protocol class when using type parameter syntax (hint: simply remove the type arguments to Protocol) (reportGeneralTypeIssues)". Maybe "Type arguments to Protocol when using type parameter syntax are redundant and must be removed".

This would greatly increase the immediate tractability of working with the error.

**Comment by erictraut** [comment]
This comes directly from the guidance in PEP 695.

> A Protocol base class with type arguments may generate a runtime error. Type checkers should generate an error in this case because the use of type arguments is not needed, and the order of type parameters for the class are no longer dictated by their order in the Protocol base class.

```python
class ClassA[S, T](Protocol): ... # OK

class ClassB[S, T](Protocol[S, T]): ... # Recommended type checker error
```

I think the current error message is quite clear. The current phrasing of the message is consistent with other similar messages in pyright, so I'm going to leave it as is. I might consider changing it in the future if I receive feedback from other pyright users, but for now you're the first and only person who has commented on this.
